File: week09/20/pr4_max_profit.py
def solution(A):
    # Implement your solution here
    if not A : return 0
    
    profit = 0
    min_val = A[0]
    for x in A :
        profit = max(profit,x-min_val)
        min_val = min(min_val,x)
    
    return profit

File: week09/20/pr4_max_profit.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;
#include<vector>
using namespace std;

int solution(vector<int> &A) {
    // Implement your solution here
    if(A.empty()) return 0;

    int profit = 0;
    int temp_min = A[0];

    for(const int x : A){
        profit = max(profit,x-temp_min);
        temp_min = min(temp_min,x);
    }
    return profit;
}
